react/components/atoms/lightbox/lightbox.js

```diff
@@ -69,10 +137,161 @@ export class Lightbox extends PureComponent {
+
+        const dx = event.nativeEvent.translationX + this.translatedX;
+        const dy = event.nativeEvent.translationY + this.translatedY;
+
+        if (Math.abs(dx) < this.translatedXTreshold) {
+            this.state.translateX.setValue(dx);
+        }
+        if (Math.abs(dy) < this.translatedYTreshold) {
+            this.state.translateY.setValue(dy);
+        }
+    };
+
+    onPanGestureEnd = event => {
+        if (event.nativeEvent.state !== State.END) return;
+
+        this.translatedX = this.state.translateX._value;
+        this.translatedY = this.state.translateY._value;
+    };
+
+    onPinchGesture = event => {
+        const pinchScale = event.nativeEvent.scale;
+        const scale = this.baseScale._value * pinchScale;
+        if (scale < this.props.minZoomScale || scale > this.props.maxZoomScale) return;
+
+        this.scaleRate.setValue(pinchScale);
+        this.lastScale = scale;
+        this.setState({ zoomed: true });
+    };
+
+    onPinchGestureEnd = event => {
+        if (event.nativeEvent.oldState !== State.ACTIVE) return;
+
+        // as the pinch is always centered we reset translation
+        // as housekeeping
+        this.resetTranslation();
+
+        this.baseScale.setValue(this.lastScale);
+        this.scaleRate.setValue(1);
+        this._setTranslateTresholds(this.lastScale);
+    };
+
+    onDoubleTap = event => {
+        if (event.nativeEvent.state !== State.ACTIVE) return;
+
+        if (!this.state.zoomed) {
+            this._doubleTapZoomIn(event.nativeEvent);
+        } else {
+            this._doubleTapZoomOut();
+        }
+    };
+
+    _doubleTapZoomIn = event => {
+        this._setTranslateTresholds(this.fullZoomedInValue);
+        const translateX = this._getDoubleTapZoomInTranslateX(event);
+        const translateY = this._getDoubleTapZoomInTranslateY(event);
+
+        this.translatedX = translateX;
+        this.translatedY = translateY;
+
+        this.setState({ zooming: true }, () => {
+            Animated.parallel([
+                Animated.timing(this.scaleRate, {
+                    toValue: this.fullZoomedInValue,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateX, {
+                    toValue: translateX,
+                    duration: this.props.translateAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateY, {
+                    toValue: translateY,
+                    duration: this.props.translateAnimationDuration,
+                    useNativeDriver: true
+                })
+            ]).start(() => {
+                const scaleRatio = this.baseScale._value * this.fullZoomedInValue;
+                this.lastScale = scaleRatio;
+                this.baseScale.setValue(this.lastScale);
+                this.scaleRate.setValue(1);

```

Comment:
Is this related to that problem you told me about? Since you are setting the scaleRate above to `this.fullZoomedInValue` and later you reset it.
Either way, why are these values not being used in the transform below?

Reply:
Yes, the redundancy is about the problem I've talked about. 
The values alter the `scale` state property